lib/model/Mode.js

```diff
@@ -0,0 +1,64 @@
+const path = require('path');
+
+const Physical = require(path.join(__dirname, 'Physical.js'));
+
+module.exports = class Mode {
+  constructor(jsonObject, fixture) {
+    this.jsonObject = jsonObject; // calls the setter
+    this.fixture = fixture; // also calls the setter
+  }
+
+  set jsonObject(jsonObject) {
+    this._jsonObject = jsonObject;
+    this._cache = {};
+  }
+
+  set fixture(fixture) {
+    this._fixture = fixture;
+    this._cache = {};
+  }
+
+
+  get name() {
+    return this._jsonObject.name; // required
+  }
+
+  get shortName() {
+    return this._jsonObject.shortName || this._jsonObject.name;
+  }
+
+  get physicalOverride() {
+    if (!('physicalOverride' in this._cache)) {
+      this._cache.physicalOverride = 'physical' in this._jsonObject ? new Physical(this._jsonObject.physical) : null;
+    }
+    return this._cache.physicalOverride;
+  }
+
+  // fixture's physical with mode's physical override applied on
+  get physical() {
+    if (!('physical' in this._cache)) {
+      if (this._fixture.physical === null || this.physicalOverride === null) {
+        this._cache.physical = this.physicalOverride || this._fixture.physical;
```

Comment:
This line is not very readable. Actually, I think we can omit the whole `if` block since null and undefined values are ignored by Object.keys.